### ISSUE TITLE
How is the CNN trained?

### ISSUE THREAD

**Issue by squidszyd** [nwojke/deep_sort#24]
Just want to know if the CNN is trained based on a weight-sharing Siamese network with triplet loss head? Or is it trained with a softmax loss where the number of classes equals to number of person IDs?

**Comment by abewley** [comment]
Sorry for the late reply.

The CNN was trained with a novel cosine softmax loss where the number of classes equal the number of IDs in a separate Re-ID training dataset. You can find code of training [here](https://github.com/nwojke/cosine_metric_learning).

**Comment by squidszyd** [comment]
wow, I don't even remember I've asked this question LOL. Thanks a lot
